docs/specifications/tests/Zone-TP/zone01.md

```diff
@@ -3,43 +3,262 @@
+There is at least one old and expired Internet-Draft that attempted to standardize
+that behavior, i.e. [draft-jabley-dnsop-missing-mname]. If the SOA MNAME
+is an empty name (".") this Test Case will not try to connect to a server behind
+it since there will never be a server behind that name, as the purpose is most
+definitely to follow that practice. Instead, a special message will be outputted.
+ 
+This Test Case will check that:
+ - the SOA MNAME contains the master name server of *Child Zone*,
+   as best as it can be determined.
+ - the SOA MNAME name server is authoritative of *Child Zone*.
+ - the SOA SERIAL of the SOA MNAME is at least equal to the ones found
+   from the name servers in the NS record set of *Child Zone*.
+   This comparison must be done following [RFC1982].
+ - the SOA MNAME name server is listed as part of the NS record set of *Child Zone*.
 
-### Special procedural requirements
+## Scope
+
+It is assumed that *Child Zone* has been tested and reported by [CONNECTIVITY01].
+This Test Case will just ignore non-responsive name servers or name servers
+not giving a correct DNS response for an authoritative name server, except
+if the name server is listed in the SOA MNAME.
+
+The syntax of the SOA MNAME for *Child Zone* is not tested in this Test
+Case, see [SYNTAX07].
+
+The consistency of the SOA MNAME between different servers of *Child Zone*
+is not tested by this Test Case, see [CONSISTENCY06].
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+
+Message Tag                   | Level   | Arguments             | Message ID for message tag
+:---------------------------- |:--------|:----------------------|:---------------------------------------------------------------------------------------
+Z01_MNAME_HAS_LOCALHOST_ADDR  | WARNING | nsname, ns_ip         | SOA MNAME name server "{nsname}" resolves to a localhost IP address ({ns_ip}).
+Z01_MNAME_IS_DOT              | NOTICE  | ns_ip_list            | SOA MNAME is specified as "." which usually means "no server". Fetched from name servers "{ns_ip_list}".
+Z01_MNAME_IS_LOCALHOST        | WARNING | ns_ip_list            | SOA MNAME name server is "localhost", which is invalid. Fetched from name servers "{ns_ip_list}".
+Z01_MNAME_IS_MASTER           | DEBUG   | ns, ns_ip_list        | SOA MNAME name server "{ns}" does appear to be master. Fetched from name servers "{ns_ip_list}".
+Z01_MNAME_MISSING_SOA_RECORD  | WARNING | ns, ns_ip_list        | SOA MNAME name server "{ns}" reponds to an SOA query with no SOA records in the answer section. Fetched from name servers "{ns_ip_list}".
+Z01_MNAME_NO_RESPONSE         | WARNING | ns, ns_ip_list        | SOA MNAME name server "{ns}" does not respond to an SOA query. Fetched from name servers "{ns_ip_list}".
+Z01_MNAME_NOT_AUTHORITATIVE   | WARNING | ns, ns_ip_list        | SOA MNAME name server "{ns}" is not authoritative for the zone. Fetched from name servers "{ns_ip_list}".
```

Comment:
As I read the message the name servers "ns_ip_list" says that "ns" is not authoritative for the zone.

I see what you want to achieve, but I think the message would be too long. If Consistency is updated it could report which name server is reporting which MNAME in case of inconsistency.